### ISSUE TITLE
U2F not supported

### ISSUE THREAD

**Issue by tsenger** [nextcloud/twofactor_u2f#96]
Hi,
I just updated to Nextcloud 13.0 and it seems the U2F app (version 1.5.1) now doesn't work anymore.
If I want to add a new token I received the message "U2F not supported" and the Google Authenticator doesn't show up anymore.
![screenshot_20180208-111114](https://user-images.githubusercontent.com/6859154/35967506-5f0d4546-0cc1-11e8-91aa-06791477767e.jpg)

I use Chrome an Android in version 63.0.3281.137. Before the update from Nextcloud 12.04 it the app worked fine. Any idea what happend?

**Comment by ChristophWurst** [comment]
We had to replace the U2F library because the old one was completely broken on newer browsers like FF. I already suspected to have a breaking change here when I did the switch to the new lib: https://github.com/nextcloud/twofactor_u2f/pull/74#issuecomment-351125123

Unfortunately, we're still waiting for the upstream library to integrate our fix to bring back U2F on Android: https://github.com/grantila/u2f-api/pull/7

**Comment by ChristophWurst** [comment]
- [ ] Blocked by https://github.com/grantila/u2f-api/pull/7

**Comment by gergelypolonkai** [comment]
I get the same problem in Chrome, both on Windows and Linux. The toast notification says “Registration failed”, but there is nothing in the browser console.

**Comment by ChristophWurst** [comment]
@gergelypolonkai which error code do you get? https://github.com/nextcloud/twofactor_u2f/blob/9c22857fd196ecef0f7bf39a9f2e840321e5a116/js/settingsview.js#L266-L287

**Comment by gergelypolonkai** [comment]
Neither. The notification on the top only says “Registration failed”.

**Comment by ChristophWurst** [comment]
Which version of the app are you using?

**Comment by gergelypolonkai** [comment]
Latest non-nighly, in NextCloud 13.0.0. I vaguely remember I tried it with the same outcome about two months ago, but I canʼt recall if it was the same or not.

Also, I have a TOTP add on installed, too. May it cause problems like this?

**Comment by ChristophWurst** [comment]
@gergelypolonkai please file a new ticket, your problem is unrelated to the one reported by @tsenger. When you fill out the issue template, let me know if you're using HTTP or HTTPS and whether your Nextcloud installation is hosted behind a reverse proxy. Thanks.

**Comment by tsenger** [comment]
Wouldn't it easier (a temporary workaround) to switch back to the version you used for Nextcloud 12 until the grantila/u2f-api is modified? I just tested a new NextCloud 12 installation with the previous version of the U2F plugin and run a U2F authentication with the mobile Chrome. Worked like charm.

**Comment by ChristophWurst** [comment]
> with the mobile Chrome. Worked like charm.

I never said Chrome didn't work. Firefox was broken with the old library as mentioned in https://github.com/nextcloud/twofactor_u2f/issues/96#issuecomment-364079205 and FF is more widely used than NFC on a mobile Chrome thus it's my priority to keep this app working on FF at least. Moreover, the old lib was cumbersome to use.

**Comment by ChristophWurst** [comment]
PR got merged: https://github.com/grantila/u2f-api/pull/7#issuecomment-369594251

**Comment by ChristophWurst** [comment]
Everyone, please help us test this updated library. You can find the pull request and the packaged application here: https://github.com/nextcloud/twofactor_u2f/pull/107#issuecomment-369604370

**Comment by ChristophWurst** [comment]
Fixed by  #116

**Comment by algoterranean** [comment]
Does this fix only work on Nextcloud 14.0? Not all of us can upgrade to 14, and this issue broke specifically on 13. Are you not planning on supporting 13?

**Comment by ChristophWurst** [comment]
> Does this fix only work on Nextcloud 14.0? Not all of us can upgrade to 14, and this issue broke specifically on 13. Are you not planning on supporting 13?

I will backport the fix soon: https://github.com/orgs/nextcloud/projects/4#card-8443455

**Comment by ChristophWurst** [comment]
Fix is in https://github.com/nextcloud/twofactor_u2f/pull/129. Please help test this backport release.

**Comment by ChristophWurst** [comment]
> Fix is in #129. Please help test this backport release.

There's a pre-packaged beta-release for you in https://github.com/nextcloud/twofactor_u2f/pull/129#issuecomment-376100832 :wink:

**Comment by algoterranean** [comment]
Thank you! I will test it later today.

**Comment by ChristophWurst** [comment]
The RC can be found at https://github.com/nextcloud/twofactor_u2f/releases/tag/1.5.2-RC1
